pkg/api/grpcServer.go

```diff
@@ -87,33 +96,27 @@ func (s *grpcServer) dispatch(ctx context.Context, dState *dispatchState, worker
 		return makeGRPCError(out)
 	}
 
-	responseBag := attribute.GetMutableBag(nil)
+	preprocResponseBag := attribute.GetMutableBag(requestBag)
```

Comment:
I added code to dump out all the attributes before invoking the normal adapters.